### ISSUE TITLE
Specification of Classic SwE with Bootstrap Fails to Run

### ISSUE THREAD

**Issue by jacob-lee** [NISOx-BDI/SwE-toolbox#102]
:: SwE v. 2.1.0
:: SPM12

I attempted to run a non-parametric analysis using the class SwE and encountered the following problems.

Case Naive
------------
Specification:
matlabbatch{1}.swe.smodel.type.classic.ss = 4;
matlabbatch{1}.swe.smodel.type.classic.dof_cl = 0; % naive

Fails with error message: `Undefined function or variable 'edf'`

Case Approx I
--------------
matlabbatch{1}.swe.smodel.type.classic.ss = 4;
matlabbatch{1}.swe.smodel.type.classic.dof_cl = 1; % approx I

Fails with error message: `degrees of freedom type still not implemented for the WB`

Analysis
---------
It appears that the bootstrap is only available for the modified SwE, as the classic SwE only offers naive and approx I as options.  However, for only one of these is the `not implemented` error raised.

Case Naive fails because there is no check for dof_type == 0. Case Approx I fails because dof_type == 1 raises non-implmented error. 

In swe_cp_WB.m, lines 2352-2379:

```matlab
switch dof_type
  case 1
    error('degrees of freedom type still not implemented for the WB')
  
  case 2
    CovcCovBc = 0;
    for g = 1:nGr
      CovcCovBc = CovcCovBc + Wg_2{g} * swe_vechCovVechV(Cov_vis(SwE.WB.iGr_Cov_vis_g==g,:), dofMat{g}, 1);
    end
    if (SwE.WB.stat == 'T')
      edf = 2 * cCovBc.^2 ./ CovcCovBc - 2;
    else
      edf = 2 * (sum(swe_duplication_matrix(nSizeCon), 1) * cCovBc).^2 ./ CovcCovBc - 2;
    end

  case 3
    CovcCovBc = 0;
    for g = 1:nGr
      CovcCovBc = CovcCovBc + Wg_3{g} * swe_vechCovVechV(Cov_vis(SwE.WB.iGr_Cov_vis_g==g,:), dofMat{g}, 2);
    end
    if (SwE.WB.stat == 'T')
      edf = 2 * cCovBc.^2 ./ CovcCovBc;
    else
      tmp = eye(nSizeCon);
      edf = (sum(swe_duplication_matrix(nSizeCon), 1) * cCovBc.^2 +...
	     (tmp(:)' * swe_duplication_matrix(nSizeCon) * cCovBc).^2) ./ CovcCovBc;
    end
end
```

**Comment by nicholst** [comment]
@jacob-lee Sorry for the delay in responding to this issue... we should soon have the man power to look into this and resolve it. (tagging @BryanGuillaume)

**Comment by BryanGuillaume** [comment]
It seems that the issue with "case Naive" has been resolved in PR#98. The issue with case Approx I is simply due to the fact that this degrees of freedom apporximation was not added to the WB. There were two reasons for this:

1.  For the classic SwE, it should match Approx III. So it is somehow redundant. 
2. For the modified SwE,  if there is no missing data, it should match Approx III and, if there are some missing data, it should perform less well than Approx III as the latter correct for a missing data bias while Approx I does not.

Thus, at the time, I did not feel useful to implement it for the WB. Note that it has been implemented for the parametric SwE because, when the toolbox was created, Approx II and Approx III had not been proposed yet and were actually proposed just before implementing the WB. The only advantage that I could think of to actually implement it for the WB would be for cases without missing data or for the classic SwE beacuse its compution should be in theory faster than Aprox III (though a benchmark would be needed to check if there is a significant difference).

@nicholst, do you feel that it is necessary to add it here? If not I feel we could close this issue.

**Comment by BryanGuillaume** [comment]
The reported issue with Approx I has been fixed in PR #137. Thus, all reported issues seem to have been fixed and I am closing this issue.
